packages/medusa/src/services/batch-job.ts

```diff
@@ -71,6 +136,243 @@ class BatchJobService extends TransactionBaseService<BatchJobService> {
+          id: result.id,
+        })
+
+      return result
+    })
+  }
+
+  async update(
+    batchJobId: string,
+    data: BatchJobUpdateProps
+  ): Promise<BatchJob> {
+    return await this.atomicPhase_(async (manager) => {
+      const batchJobRepo: BatchJobRepository = manager.getCustomRepository(
+        this.batchJobRepository_
+      )
+
+      let batchJob = await this.retrieve(batchJobId)
+
+      const { context, ...rest } = data
+      if (context) {
+        batchJob.context = await this.validateBatchContext(
+          batchJob.type,
+          context
+        )
+      }
+
+      Object.keys(rest)
+        .filter((key) => typeof rest[key] !== `undefined`)
+        .forEach((key) => {
+          batchJob[key] = rest[key]
+        })
+
+      batchJob = await batchJobRepo.save(batchJob)
+
+      await this.eventBus_
+        .withTransaction(manager)
+        .emit(BatchJobService.Events.UPDATED, {
+          id: batchJob.id,
+        })
+
+      return batchJob
+    })
+  }
+
+  protected async updateStatus(
+    batchJobOrId: BatchJob | string,
+    status: BatchJobStatus
+  ): Promise<BatchJob | never> {
+    return await this.atomicPhase_(
+      async (transactionManager: EntityManager) => {
+        let batchJob: BatchJob = batchJobOrId as BatchJob
+        if (typeof batchJobOrId === "string") {
+          batchJob = await this.retrieve(batchJobOrId)
+        }
+
+        const { entityColumnName, eventType } =
+          this.batchJobStatusMapToProps.get(status) || {}
+
+        if (!entityColumnName || !eventType) {
+          throw new MedusaError(
+            MedusaError.Types.INVALID_DATA,
+            `Unable to update the batch job status from ${batchJob.status} to ${status}. The status doesn't exist`
+          )
+        }
+
+        batchJob[entityColumnName] = new Date()
+
+        const batchJobRepo = transactionManager.getCustomRepository(
+          this.batchJobRepository_
+        )
+        batchJob = await batchJobRepo.save(batchJob)
+
+        this.eventBus_.withTransaction(transactionManager).emit(eventType, {
+          id: batchJob.id,
+        })
+
+        return batchJob
+      }
+    )
+  }
+
+  async confirm_processing(
+    batchJobOrId: string | BatchJob
+  ): Promise<BatchJob | never> {
+    return await this.atomicPhase_(
+      async (transactionManager: EntityManager) => {
+        let batchJob: BatchJob = batchJobOrId as BatchJob
+        if (typeof batchJobOrId === "string") {
+          batchJob = await this.retrieve(batchJobOrId)
+        }
+
+        if (!batchJob.dry_run) {
+          return batchJob
+        }
+
+        if (batchJob.status !== BatchJobStatus.READY) {
+          throw new MedusaError(
+            MedusaError.Types.NOT_ALLOWED,
+            "Cannot confirm processing for a batch job that is not ready"
+          )
+        }
+
+        this.eventBus_
+          .withTransaction(transactionManager)
+          .emit(BatchJobService.Events.PROCESSING, {
```

Comment:
todo: if you like the idea of adding an additional status (confirmed) this would become
```suggestion
          .emit(BatchJobService.Events.CONFIRMED, {
```